lib/streamlit/elements/arrow_altair.py

```diff
@@ -1171,20 +1171,6 @@ def _get_opacity_encoding(
     return None
 
 
-def _get_scale(df: pd.DataFrame, column_name: str | None) -> alt.Scale:
-    import altair as alt
-
-    # Set the X and Y axes' scale to "utc" if they contain date values.
-    # This causes time data to be displayed in UTC, rather the user's local
-    # time zone. (By default, vega-lite displays time data in the browser's
-    # local time zone, regardless of which time zone the data specifies:
-    # https://vega.github.io/vega-lite/docs/timeunit.html#output).
-    if _is_date_column(df, column_name):
-        return alt.Scale(type="utc")
-
-    return alt.Scale()
```

Comment:
We can safely remove this due to the change of marshaling in the arrow format. This change was originally with the [original implementation](https://github.com/streamlit/streamlit/pull/621) where we set this up in the first place.

1. They only apply to built-in charts where they are generated.
2. I verified the change worked with the cypress images from the original PR before migrating them to the playwright.
3. The marshaling of date columns to arrow essentially assume they are in UTC format (similar to the other scenarios where we have managed the arrow parsing)

Reply:
@LukasMasuch Setting the scale only sets the scale in vega-lite specification. So, the datetimes continue to be sent down as datetimes without timezones, but the scale is specified, and the question becomes, are the dates in UTC or not? With no timezone, I believe the answer doesn't matter, we want the dates to display the same as what is written in Python. VegaLite assumes the dates are in local timezones, so the offset remains needed.

Perhaps the TLDR is `alt.Scale(type="utc")` is only useful IF a timezone is specified. Otherwise, it's effectively ignored.

Reply:
I played a bit around with the current state and it seems to work as expected 👍  I'm still a bit confused to what `alt.Scale(type="utc")` is actually supposed to do... in my experiments there wasn't really any significant difference to `alt.Scale(type="time")`. And from looking at the description [here](https://vega.github.io/vega-lite/docs/scale.html#time) I would have assumed that it has some kind of impact on all types of date objects / timestamps. But this doesn't seem to be the case. 

Let's merge and see if anyone complains about anything time-scale related.